python/lsst/meas/extensions/multiprofit/plots.py

```diff
@@ -0,0 +1,361 @@
+# This file is part of meas_extensions_multiprofit.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+from abc import ABC, abstractmethod
+import astropy.table
+import astropy.units as u
+from lsst.multiprofit.plots import bands_weights_lsst, plot_model_rgb
+import matplotlib.axes
+import matplotlib.figure
+import matplotlib.pyplot as plt
+import numpy as np
+import pydantic
+from pydantic.dataclasses import dataclass
+from .rebuild_coadd_multiband import PatchCoaddRebuilder
+from typing import Any, Iterable
+
+__all__ = [
+    "ObjectTableBase",
+    "TruthSummaryTable",
+    "ObjectTable",
+    "ObjectTableCModel",
+    "ObjectTableMultiProFit",
+    "ObjectTablePsf",
+]
+
+Figure = matplotlib.figure.Figure
+Axes = matplotlib.axes.Axes | Iterable[matplotlib.axes.Axes]
+FigureAxes = tuple[Figure, Axes]
+
+
+@dataclass(frozen=True, config=pydantic.ConfigDict(arbitrary_types_allowed=True))
```

Comment:
Naive question, but what's the gain in using a pydantic dataclass with lots of explicit pydantic code over using `pydantic.BaseModel` directly? The `@dataclass` decorator is confusing to me at first glance when this is clearly a data model.

Reply:
It's really just force of habit as I started using pydantic dataclasses to replace `dataclasses.dataclass`, which is what they were made for. I don't think I've ever actually extended `BaseModel`.

Looking into this there is some discussion of the small differences in behaviour between dataclass and BaseModel here: https://github.com/pydantic/pydantic/issues/710. I don't think any of those issues have impacted me yet, so I'm uncertain as to whether it's best to use one or the other. In the same way that using dataclass over BaseModel looks strange to someone familiar with pydantic, using dataclass should look familiar to anyone who only uses standard library dataclasses (though I suppose by that argument I should use `from pydantic import Field` rather than `pydantic.Field`).

Reply:
My main comment would be that if you are wanting to switch to pydantic for new code it would be better to use BaseModel -- the ConfigDict you are using is also non-standard. The pydantic dataclasses support is to ease migration of existing code and not something that we should be doing for new code unless we are worried that we might drop pydantic in the future. cc/ @TallJimbo

Reply:
Fair enough, I changed to `BaseModel` and it was painless.